kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyConfigSecret.java

```diff
@@ -83,6 +96,22 @@ static String secretName(KafkaProxy primary) {
         return primary.getMetadata().getName();
     }
 
+    public static List<Volume> secureVolumes(ManagedDependentResourceContext managedDependentResourceContext) {
+        Set<Volume> volumes = managedDependentResourceContext.get(ProxyConfigSecret.SECURE_VOLUME_KEY, Set.class).orElse(Set.of());
```

Comment:
A refactor for later: it would be nice if these kind of functions for working with the managed context were wrapped up in another class, so that ProxyConfigSecret and ProxyDeployment could call methods on something without both having to be aware of how to work with the context sets.

Maybe we could introduce our own superclass extending `CRUDKubernetesDependentResource` so that they override:

```
protected Secret desired(KafkaProxy primary,
                             OurEnhancedContext context) 
```

Reply:
we already have `SharedKafkaProxyContext`, but using this superclass might help prevent it getting forgotten/lost